rotkehlchen/exchanges/binance.py

```diff
@@ -286,13 +286,15 @@ def api_query(self, method: str, options: Optional[Dict] = None) -> Union[List,
                         msg,
                     ))
             elif response.status_code == 429:
-                if backoff > self.backoff_limit:
-                    break
-                # Binance has limits and if we hit them we should backoff
-                # https://github.com/binance-exchange/binance-official-api-docs/blob/master/rest-api.md#limits
-                log.debug('Got 429 from Binance. Backing off', seconds=backoff)
-                gevent.sleep(backoff)
-                backoff = backoff * 2
+                # Binance has limits and if we hit them we should backoff.
+                # A Retry-After header is sent with a 418 or 429 responses and
+                # will give the number of seconds required to wait, in the case
+                # of a 429, to prevent a ban, or, in the case of a 418, until
+                # the ban is over.
+                # https://binance-docs.github.io/apidocs/spot/en/#limits
+                retry_after = int(response.headers.get('retry-after', '0'))
+                log.debug('Got 429 from Binance. Backing off', seconds=retry_after)
+                gevent.sleep(retry_after)
```

Comment:
I couldn't find any inconvenient about `gevent.sleep(0)`. If it is not right, I can do a:
```python
if retry_after > 0:
    gevent.sleep(retry_after)
```